File: python/interview_preparation_kit/problem_4_jumping_on_the_clouds.py
import time

def jumpingOnClouds(c):
    jumps = 0
    position = 0
    n = len(c)
    while position < n -1:
        if position + 2 < n and c[position + 2] == 0:
            position += 2
        else:
            position += 1
        jumps += 1
    return jumps


if __name__ == "__main__":
    start_time = time.time()
    clouds = [0, 0, 0, 0, 0, 0, 1, 0, 0]
    result = jumpingOnClouds(clouds)
    elapsed_time = time.time() - start_time
    print(result)
    print(f"--- {elapsed_time:.4f} seconds ---")

File: JavaScript/interview_preparation_kit/problem_4_jumping_on_the_clouds.js
/* function jumpingOnClouds(c) {
  let jumpCount = 0;
  while (c.length > 1) {
    if (c[2] === 0) {
      c.shift();
    }
    c.shift();
    jumpCount++;
  }
  return jumpCount; 
} */

function jumpingOnClouds(c) {
  let jumpCount = 0;
  let position = 0;
  while (position < c.length - 1) {
    position += (c[position + 2] === 0) ? 2 : 1;
    jumpCount++;
  }
  return jumpCount; 
}

const c = [0, 0, 0, 0, 0, 1, 0];
console.time("Execution Time");
const result = jumpingOnClouds(c);
console.timeEnd("Execution Time");
console.log(result);
